grpc/src/main/java/com/linecorp/armeria/server/grpc/ProtoReflectionServiceInterceptor.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.grpc.Context;
+import io.grpc.Contexts;
+import io.grpc.InternalServer;
+import io.grpc.Metadata;
+import io.grpc.Server;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerServiceDefinition;
+
+final class ProtoReflectionServiceInterceptor implements ServerInterceptor {
+
+    static final ProtoReflectionServiceInterceptor INSTANCE = new ProtoReflectionServiceInterceptor();
+
+    @Nullable
+    private static Server dummyServer;
+
+    private ProtoReflectionServiceInterceptor() {}
+
+    @Override
+    public <I, O> Listener<I> interceptCall(ServerCall<I, O> call, Metadata headers,
+                                            ServerCallHandler<I, O> next) {
+        if (dummyServer == null) {
+            synchronized (INSTANCE) {
+                dummyServer = newDummyServer();
+            }
+        }
```

Comment:
How about creating eagerly?

Reply:
Oh, I think we can create the dummy when `serviceAdded(...)` is called and use that in this interceptor.